Source/BoundaryConditions/WarpX_PEC.H

```diff
@@ -295,6 +309,345 @@ using namespace amrex;
+                           const amrex::IntVect & ijk_vec,
+                           amrex::Array4<amrex::Real> const& Jfield,
+                           const amrex::IntVect & is_nodal,
+                           amrex::GpuArray<int, 3> const& fbndry_lo,
+                           amrex::GpuArray<int, 3> const& fbndry_hi,
+                           amrex::GpuArray<ParticleBoundaryType, 3> const& pbndry_lo,
+                           amrex::GpuArray<ParticleBoundaryType, 3> const& pbndry_hi )
+    {
+        amrex::IntVect ijk_mirror;
+        // Loop over all dimensions
+        for (int idim = 0; idim < AMREX_SPACEDIM; ++idim) {
+            ijk_mirror = ijk_vec;
+            // Loop over sides, iside = 0 (lo), iside = 1 (hi)
+            for (int iside = 0; iside < 2; ++iside) {
+                // check if the boundary in this direction is a PEC, otherwise
+                // continue to the next direction
+                if (!( (iside == 0 )
+                    ? is_boundary_PEC(fbndry_lo, idim)
+                    : is_boundary_PEC(fbndry_hi, idim) )) continue;
+
+#if (defined WARPX_DIM_1D_Z)
+                // For 1D : icomp=0 and icomp=1 (Ex and Ey are tangential to the z boundary)
+                //          The logic below ensures that the flags are set right for 1D
+                const bool is_tangent_to_PEC = ( ( icomp == idim+2) ? false : true );
+#elif (defined WARPX_DIM_XZ) || (defined WARPX_DIM_RZ)
+                // For 2D : for icomp==1, (Ey in XZ, Etheta in RZ),
+                //          icomp=1 is tangential to both x and z boundaries
+                //          The logic below ensures that the flags are set right for 2D
+                const bool is_tangent_to_PEC = ( (icomp == AMREX_SPACEDIM*idim)
+                                             ? false : true );
+#else
+                const bool is_tangent_to_PEC = ( ( icomp == idim) ? false : true );
+#endif
+
+                // Mirror guard cell outside the domain by "ig" number of cells, in
+                // direction, idim.
+                // Note: For nodal fields ig=0 is on the boundary and will be
+                // handled in `SetJFieldOnPEC` below. For cell-centered
+                // fields ig=0 is the cell center next to the PEC (inside the
+                // domain) and therefore should not be treated either.
+                const int ig = ( (iside == 0)
+                                ? (ijk_vec[idim] - dom_lo[idim] + (1 - is_nodal[idim]))
+                                : (dom_hi[idim] + 1 - ijk_vec[idim]) );
+
+                if (ig > 0 && ig <= nguard[idim]) {
+
+                    // Mirror location outside the domain by "ig" number of cells
+                    // across PEC boundary in direction, idim, and side, iside
+                    ijk_mirror[idim] = ( (iside == 0)
+                                     ? (dom_lo[idim] - ig)
+                                     : (dom_hi[idim] + ig + is_nodal[idim]));
```

Comment:
@roelof-groenewald here as well right? 
`dom_hi + 1 + ig`

Reply:
I just pushed a change where the `ig` and `ijk_mirror` calculation is made uniform for all the functions in `WarpX_PEC.H`, so hopefully that will take care of all those instances.